File: backend/core/src/apac_core/domain/entities/apac_batch.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime, date
from apac_core.domain.entities.city import City
from apac_core.domain.value_objects.validity import Validity
from apac_core.domain.entities.apac_request import ApacRequest

class ApacBatch(BaseModel):
    batch_number: str
    city: City
    validity: Validity
    apac_request: Optional[ApacRequest] = None
    export_date: Optional[date] = None
    id: Optional[int] = None

    def is_available(self, now: datetime) -> bool:
        return self.apac_request is None and not self.validity.is_expired(now)

    def set_apac_request(self, apac_request: ApacRequest):
        self.apac_request = apac_request

File: frontend/src/app/extracao/schemas/apacBatch.ts
import { ApacRequest } from "@/app/solicitar/apacRequest/schemas/apacRequest";
import { City } from "@/shared/schemas";


interface Validity {
    expire_in: Date;
    created_in: string;
}

export interface ApacBatch {
    batch_number: string;
    city: City;
    validity: Validity;
    apac_request: ApacRequest;
    export_date?: Date;
    id: number;
}